### PR TITLE
Add example app

### LINKED ISSUE #8: Example app
# Requirements

Create an app that runs locally and can trigger the interface methods with various inputs. Can be  console or UI.

### PR DESCRIPTION
**Pull request by alex-tpom6oh**
Fixes #8

### CODE REVIEW

**Comment by hugihlynsson** [comment]
Should we merge this?

**Comment by alex-tpom6oh** [comment]
Would like to get a review here before merging, would you be up for reviewing @hugihlynsson ?

**Comment by hugihlynsson** [comment]
I'll do my best!

File: lib/avo_inspector.dart [hunk 1]
```diff
@@ -19,7 +19,8 @@ class AvoInspector {
       required this.appName});
 
   List<Map<String, dynamic>> trackSchemaFromEvent(
-      String eventName, Map<String, dynamic> eventProperties) {
+      {required String eventName,
+      required Map<String, dynamic> eventProperties}) {
```
Comment (hugihlynsson) [inline lib/avo_inspector.dart:23]: Is this a breaking change? Do we want to include it in a changelog?

File: example_app/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      theme: ThemeData(
+        // This is the theme of your application.
+        //
+        // Try running your application with "flutter run". You'll see the
+        // application has a blue toolbar. Then, without quitting the app, try
+        // changing the primarySwatch below to Colors.green and then invoke
+        // "hot reload" (press "r" in the console where you ran "flutter run",
+        // or simply save your changes to "hot reload" in a Flutter IDE).
+        // Notice that the counter didn't reset back to zero; the application
+        // is not restarted.
```
Comment (hugihlynsson) [inline example_app/lib/main.dart:38]: Should we remove this comment and the rest of the default ones to make the Avo related code stand out more?

File: example_app/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  Widget build(BuildContext context) {
+    AvoInspector.enableLogs(true);
+
+    avoInspector = AvoInspector(
+        apiKey: "my api key",
+        env: AvoInspectorEnv.dev,
+        appVersion: "",
+        appName: "");
+    avoInspector
+        ?.trackSchemaFromEvent(eventName: "App Open", eventProperties: {});
```
Comment (hugihlynsson) [inline example_app/lib/main.dart:25]: Should we add a comment on what's happening and why it's here?

File: example_app/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  void _incrementCounter() {
+    setState(() {
+      // This call to setState tells the Flutter framework that something has
+      // changed in this State, which causes it to rerun the build method below
+      // so that the display can reflect the updated values. If we changed
+      // _counter without calling setState(), then the build method would not be
+      // called again, and so nothing would appear to happen.
+      _counter++;
+    });
+    avoInspector?.trackSchemaFromEvent(eventName: "Counter++", eventProperties: {"counter" : _counter});
```
Comment (hugihlynsson) [inline example_app/lib/main.dart:76]: Should we add a comment here?

File: example_app/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
+- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
+
+For help getting started with Flutter, view our
+[online documentation](https://flutter.dev/docs), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
```
Comment (hugihlynsson) [inline example_app/README.md:16]: Should we add a comment here on what changes were made to the default project to make Avo work?

**Review by hugihlynsson — APPROVED** [review]
Had a few suggestions but it looks great to me!

**Review by logason — APPROVED** [review]

**Pull request merged**